create-infra.yml

```diff
@@ -75,3 +75,100 @@
       ansible.builtin.import_role:
         name: "libvirt_manager"
         tasks_from: "deploy_layout.yml"
+
+    - name: Gather ansible_user_id
+      ansible.builtin.setup:
+        gather_subset:
+          - user_id
+          - user_dir
+
+    - name: Read inventory groups
+      ansible.builtin.find:
+        paths: "{{ ansible_user_dir }}/ci-framework-data/reproducer-inventory/"
+        file_type: file
+        patterns: '*.yml,*.yaml'
+      register: "_all_groups"
+
+    - name: Read groups inventory
+      ansible.builtin.slurp:
+        path: "{{ item.path }}"
+      register: _reproducer_inventory_files
+      loop: "{{ _all_groups.files }}"
+
+    - name: Set groups fact
+      when: "'all' != _group_name"
+      vars:
+        _content: "{{ _inventory_file.content | b64decode | from_yaml }}"
+        _group_name: "{{ _content | list | first }}"
+        _nodes: "{{ _content[_group_name]['hosts'].keys() }}"
+        _group_nodes: >-
+          {%- set group = {} -%}
+          {%- set _ = group.update({_group_name: _nodes}) -%}
+          {{ group }}
+      ansible.builtin.set_fact:
+        _vm_groups: >-
+          {{ _vm_groups | default({}) | combine(_group_nodes) }}
+      loop: "{{ _reproducer_inventory_files.results }}"
+      loop_control:
+        loop_var: "_inventory_file"
```

Comment:
Parsing an Ansible inventory directly, is never a good idea, there are more than one way of adding an instance to a group :) I'd suggest to use https://docs.ansible.com/ansible/latest/reference_appendices/special_variables.html#term-groups. That is the one we use for the networking mapper.

Reply:
yeah, I copied this from here https://github.com/openstack-k8s-operators/ci-framework/blob/b794aa220f6290399c6314ab7a85a11bf2f9caed/deploy-osp-adoption.yml#L46 since I saw the `_vm_groups` fact was used when starting vms here: https://github.com/openstack-k8s-operators/ci-framework/blob/b794aa220f6290399c6314ab7a85a11bf2f9caed/roles/adoption_osp_deploy/tasks/main.yml#L35

I changed it to drop the entried read of reproducer inventory files.
*lmknwyt*